src/lib.rs

```diff
@@ -790,9 +790,10 @@ pub mod prelude {
 /// # Panics
 ///
 /// Panics if `description` doesn't start with `/`.
-pub fn route<S, B>(description: &str, service: S) -> Route<S, EmptyRouter<S::Error>>
+pub fn route<S, B, P>(description: P, service: S) -> Route<S, EmptyRouter<S::Error>>
```

Comment:
should `description` be changed to `path`? Description is usually inert metadata but `P` in this case is used for routing logic.